Percona toolkit release v3.3.2

percona-csalguero opened the pull request:
This is **WIP**.
- Updated the changelog to reflect the current status and what's need to be completed for release.
- Updated 3.3.1 release date because it was not set by the release script
- Sorted previous releases ticket to maintain our sorting rules. 

Missing things:
- [ ] Update Go dependencies

percona-csalguero committed:
Updated changelog for (WIP) 3.3.2

vinzlj commented:
Hi @percona-csalguero, any update on this release?

I'm looking forward to trying the drop_swap method with MySQL 8+.

Thanks!

percona-csalguero commented:
I cannot tell you a release date, but you can download the latest versions from the 3.x branch. 

Regards

percona-csalguero committed:
Merge branch '3.x' into release-3.3.2

benlangfeld commented on Changelog:
```diff
@@ -1,9 +1,32 @@
 
- * Improvement PT-1974: Support fingerprinting for --print in pt-kill (Thanks Iwo Panowicz)
- * Fixed bug   PT-1966: Fixed test (Thanks @yoku0825)
+v3.3.2 WIP release date not set yet
```
Doesn't a release date need to be set before this is mergeable?

percona-csalguero replied:
No. The build script will set the release date

rnovikovP commented:
@percona-csalguero, I see in PR for 3.3.2  in TODO issues we plan for 3.4.0 release. 
So I'm not very clear what we are trying to achieve with this PR

percona-csalguero committed:
Updated changelog

percona-csalguero committed:
Merge branch 'release-3.3.2' of percona.github.com:percona/percona-toolkit into release-3.3.2

rnovikovP commented on Changelog:
```diff
@@ -1,9 +1,31 @@
- * Fixed bug   PT-1966: Fixed test (Thanks @yoku0825)
+v3.3.2 WIP release date not set yet
+
+ * Improvement PT-1417: Inconsistent creation of toolkit tables
```
please see https://confluence.percona.com/display/DOC/Release+Notes+Template+and+Authoring+Guide 
This title needs to be about the Value we added, and it's hard to get it from there now.

Something like "Automated creation of tables in case of .... to avoid ... "

rnovikovP commented on Changelog:
```diff
@@ -1,9 +1,31 @@
+ * Fixed bug   PT-1966: Test no_drop_no_swap for the pt-online-schema-change is broken (Thanks Tsubasa Tanaka)
+ * Fixed bug   PT-1974: Support fingerprinting for --print in pt-kill
+ * Fixed bug   PT-1983:	pt-summary missing one DIMM	
+ * Admin task  PT-2009: Update Go Lang dependencies
+ * Admin task  PT-2011:	Make build reproducible
```
This is an Admin task. We usually do not include them in RN. 
 They need to be removed or converted to another appropriate type if this  a bug or improvement.

denisok replied:
do we do RNs from Changelog? this is a changelog....

percona-csalguero replied:
Yes. This is only the changelog to track what we are going to include in next release.

rnovikovP commented:
For changelog - let's just clean up the first item description as this is very confusing. 
And I'm not clear  why we including items we  still have IN REVIEW  (status like PT-1627)

percona-csalguero commented:
> For changelog - let's just clean up the first item description as this is very confusing. And I'm not clear why we including items we still have IN REVIEW (status like PT-1627)

I've updated the changelog and PT-1627 and PT-1398 are merged now.

benlangfeld commented:
Hey guys. My team just got bit by [this](https://github.com/percona/percona-toolkit/pull/474) again yesterday. What is blocking this release?

denisok commented:
@benlangfeld looks like we are now targeting 3.4 instead, there are more fixes and etc there:
https://jira.percona.com/projects/PT/versions/15001
https://jira.percona.com/projects/RM/issues/RM-930

what quality do you need that fix to e, maybe we could provide RC or some custom build?
CC @rnovikovP

benlangfeld commented:
@denisok The problem is just the difficulty of coordinating installation of custom builds for all members of a large team, vs tracking the latest release. The fix I care about was merged 4 months ago; it doesn't seem to be necessary to have such a long release cycle for such a simple thing.

benlangfeld commented:
@denisok @percona-csalguero Is it possible to just cut a release? It seems we just keep waiting to include more and more stuff and there's no cut-off.

BupycHuk committed:
PMM-9337 Fix build (#531)

percona-csalguero committed:
Ran go mod tidy

percona-csalguero committed:
Merge branch 'release-3.3.2' of percona.github.com:percona/percona-toolkit into release-3.3.2

percona-csalguero committed:
Merge branch '3.x' into release-3.3.2

percona-csalguero committed:
Fixed merge conflicts

BupycHuk committed:
PMM-9337 Go mod tidy

ryantology commented:
I would like to also request a bump to get this pr released: https://github.com/percona/percona-toolkit/pull/474
If it is possible to get a minor revision or RC that would be great. We are able to work around this but it does cause confusion for the team when they don't have the custom build.

percona-csalguero commented:
Hi,
Do you need deb or rpm packages?
If not, you can download the file from the master branch. 

Regards

benlangfeld commented:
> Hi, Do you need deb or rpm packages? If not, you can download the file from the master branch.
> 
> Regards

Hey @percona-csalguero . I think it would be really helpful if someone could explain why Percona is so resistant to releasing a new version publicly after so much time has passed. Those of us who consume these things are having a really hard time understanding what the problem is, given it seems we have a super simple solution: pull the trigger on the new version.

percona-csalguero commented:
> > Hi, Do you need deb or rpm packages? If not, you can download the file from the master branch.
> > Regards
> 
> Hey @percona-csalguero . I think it would be really helpful if someone could explain why Percona is so resistant to releasing a new version publicly after so much time has passed. Those of us who consume these things are having a really hard time understanding what the problem is, given it seems we have a super simple solution: pull the trigger on the new version.

Hi. We are working in a new release and the idea is to release it soon.

denisok commented:
@benlangfeld https://github.com/percona/percona-toolkit/pull/539 should progress faster

percona-csalguero committed:
Merge branch '3.x' into release-3.3.2

percona-csalguero committed:
Fix Go programs build (#528)

* Fix Go programs build

It was failing due to Go version change.
- Added go mod tidy into the Makefile
- Updated MongoDB driver to version 1.8.1

* Fixed conflicts

* Fixed merge conflicts
